api/src/schema/resolvers.js

```diff
@@ -43,11 +43,9 @@ const resolvers = {
 
       if (filter) {
         let value = filter.value
-        // if the value is a number
-        // - optionally starting with a minus sign
-        // - containing zero or one decimal places
-        // convert it to a float instead of a string
-        if (filter.value.match(/^-?\d+\.?\d+$/)) {
+        // if the value is either an integer or a float, optionally starting
+        // with a minus sign convert it to a float instead of a string
+        if (filter.value.match(/^-?[0-9]+$|^-?\d+\.?[0-9]*$/)) {
```

Comment:
Two things:

1. we should probably standardize on either `[0-9]` or `\d` for digits and make them all this way.

2. the problem with the last half of the expression (`^-?\d+\.?[0-9]*`) is that if you make the first digit optional, you would allow values like `-.1` or `.1`, whereas if you make the trailing digit optional, you would allow values like `-1.` or `1.`. Having said that, `parseFloat` seems to work fine with either. 

***

I don't know if the first part of the OR is doing anything important. If we just changed the original expression to `/^-?\d+\.?\d*$/`, we would have the same result.

My opinion: just change the original expression to either  `/^-?\d*\.?\d+$/` (accepts `-.1`) or  `/^-?\d+\.?\d*$/`  (accepts `-1.`) and write a test that confirms correct behaviour.